hazelcast/include/hazelcast/util/RuntimeAvailableProcessors.h

```diff
@@ -63,7 +65,7 @@ namespace hazelcast {
         private:
 
             // number of available processors currently configured
-            static util::AtomicInt currentAvailableProcessors;
+            static std::atomic<int> currentAvailableProcessors;
```

Comment:
when doing this change you have to set the variable to 0 in constructor so that the behaviour is preserved.

Reply:
that's a static variable, you can't initialize it here. it gets initialized in https://github.com/hazelcast/hazelcast-cpp-client/blob/a79088b7ca0cbdfed3ac1f3421c563b5f5e2c972/hazelcast/src/hazelcast/util/util.cpp#L1154.
By the way, what is the point of this variable to be an atomic? As far as I'm aware, it gets initialized only once and never changed later.
Also, to be safe, I first deleted default ctor of AtomicInt, zero initialized necessary places, compiled the code, and then finally replaced every AtomicInt with std::atomic<int>. So, there shouldn't be any atomic int variable left uninitialized.

Reply:
Ok, if it is static, and it is being initialized, it is OK. I think we can completely get rid of this class and use https://en.cppreference.com/w/cpp/thread/thread/hardware_concurrency instead.